### PR TITLE
Feature/qa content api

### LINKED ISSUE #143: NODE - QAContents API
__QAContents API -> /qacontents__

Allowed Roles | Description | Method | Endpoint
-- | -- | -- | --
JF | QAContents - Find All | GET | /
All | QAContents - Find One | GET | /:content_id
JF | QAContents - Create | POST | /create
JF | QAContents - Edit | PATCH | /edit/:contents_id
JF | QAContents - Change Status | PATCH | /change_status/:question_id

- `All` means any authenticated user

### Checklist
- [x] Find All
- [x] Find One
- [x] Create

Field | Type | Optional
-- | -- | --
japanese_question | string | FALSE
select0 | string | TRUE
select1 | string | TRUE
select2 | string | TRUE
select3 | string | TRUE
answer_select | string | TRUE
japanese_response | string | TRUE

- [x] Edit (Same data as Create)
- [x] Change Status

*Integrate AWS S3*

Reference:
- [Database Design Alpha - qacontents sheet](https://docs.google.com/spreadsheets/d/1lUj5XLwv89rzNixoq-NBfGRep3ZfR4faLniPbmiImHc/edit#gid=984608782)

### PR DESCRIPTION
**Pull request by sbenemerito**
Closes #143

### CODE REVIEW

File: backend_node/controllers/qacontent.controller.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+};
+
+exports.findOne = (req, res) => {
+  QAContent.findOne({
+    where: {
+      id: req.params.id,
+      is_active: true
+    }
+  })
+    .then(qacontent => {
```
Comment (hugh-n) [inline backend_node/controllers/qacontent.controller.js]: Remove two spaces from the indentation here.

File: backend_node/controllers/qacontent.controller.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  QAContent.create({
+    japanese_question: req.body.japanese_question,
+    select0: req.body.select0,
+    select1: req.body.select1,
+    select2: req.body.select2,
+    select3: req.body.select3,
+    answer_select: req.body.answer_select,
+    japanese_response: req.body.japanese_response
+  })
+    .then(qacontent => {
```
Comment (hugh-n) [inline backend_node/controllers/qacontent.controller.js]: Also here.

File: backend_node/controllers/qacontent.controller.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    });
+};
+
+exports.changeStatus = (req, res) => {
+  QAContent.findOne({
+    where: {
+      id: req.params.id
+    }
+  })
+    .then(qacontent => {
```
Comment (hugh-n) [inline backend_node/controllers/qacontent.controller.js]: Also here.

File: backend_node/controllers/qacontent.controller.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        res.status(404).json({
+          success: false,
+          msg: "No qacontent with this ID found!"
+        });
+      }
+
+      qacontent.update({
+          is_active: !qacontent.is_active
+      })
+        .then(qacontent => {
```
Comment (hugh-n) [inline backend_node/controllers/qacontent.controller.js]: And here

**Comment by sbenemerito** [comment]
QAContents API not needed (consulted with Hugh-san)

**Pull request closed without merging**